// @expo/eas-build-job/dist/generic.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Generic = void 0;
const zod_1 = require("zod");
const common_1 = require("./common");
var Generic;
(function (Generic) {
    const BuilderEnvironmentSchemaZ = zod_1.z.object({
        image: zod_1.z.string(),
        node: zod_1.z.string().optional(),
        yarn: zod_1.z.string().optional(),
        pnpm: zod_1.z.string().optional(),
        bun: zod_1.z.string().optional(),
        expoCli: zod_1.z.string().optional(),
        env: zod_1.z.record(zod_1.z.string()),
        // Linux
        ndk: zod_1.z.string().optional(),
        // macOS
        bundler: zod_1.z.string().optional(),
        fastlane: zod_1.z.string().optional(),
        cocoapods: zod_1.z.string().optional(),
    });
    Generic.JobZ = zod_1.z.object({
        projectArchive: common_1.ArchiveSourceSchemaZ,
        customBuildConfig: zod_1.z.object({
            path: zod_1.z.string(),
        }),
        secrets: zod_1.z.object({
            robotAccessToken: zod_1.z.string(),
            environmentSecrets: zod_1.z.array(common_1.EnvironmentSecretZ),
        }),
        expoDevUrl: zod_1.z.string().url(),
        builderEnvironment: BuilderEnvironmentSchemaZ,
    });
})(Generic || (exports.Generic = Generic = {}));
